tests/benchmarks/mbedtls/src/benchmark.c

```diff
@@ -118,7 +118,7 @@ void mbedtls_set_alarm(int seconds)
 {
 	mbedtls_timing_alarmed = 0;
 
-	k_delayed_work_submit(&mbedtls_alarm, K_SECONDS(seconds));
+	k_work_schedule(&mbedtls_alarm, K_SECONDS(seconds));
```

Comment:
The default replacement is `k_work_reschedule`.  

The difference is described [here](https://docs.zephyrproject.org/latest/reference/kernel/threads/workqueue.html#submitting-a-delayable-work-item) and in the Best Practices section of #33104 Use the Right Scheduling Operation.

The test is pretty obscurely written, but it appears that some loops exit either when the work handler sets the volatile flag, or when an error is returned.  It's not clear that the code correctly handles an early exit: the alarm may remain set from a previous (possibly failed) test, and `schedule` will not update it to the new deadline.

So this could be switched to `reschedule`, but it would be better if every test that used the alarm would end with a `k_work_delayable_cancel_sync()` to ensure that it's flushed.  Or, if the test can exit early, do the sync cancel before setting the alarm.
If you know the work item is not pending at all points where it's scheduled, then the difference doesn't matter,

Reply:
If you put the `k_work_cancel_delayed_sync` into the macros `TIME_AND_TSC` and `TIME_PUBLIC` so that it is processed on all paths following `mbedtls_set_alarm()` then it would be correct to leave this `schedule` rather than change it to `reschedule`.

Reply:
I'm guessing you mean k_work_cancel_sync()?

Reply:
k_work_cancel_delayable_sync()